src/gui/attributeformconfig/qgsattributeformcontaineredit.h

```diff
@@ -19,24 +19,38 @@
 // We don't want to expose this in the public API
 #define SIP_NO_FILE
 
-#include <QWidget>
-
+#include "qgsattributesformmodel.h"
+#include "qgis_gui.h"
 #include "ui_qgsattributeformcontaineredit.h"
 
-#include "qgis_gui.h"
+#include <QWidget>
 
-class QTreeWidgetItem;
 
 /**
- * Widget to edit a container (tab or group box) of a form configuration
+ * Widget to edit a container (tab, group box, or row) of a form configuration.
+ *
  * \since QGIS 3.14
  */
 class GUI_EXPORT QgsAttributeFormContainerEdit : public QWidget, private Ui_QgsAttributeFormContainerEdit
 {
     Q_OBJECT
 
   public:
-    explicit QgsAttributeFormContainerEdit( QTreeWidgetItem *item, QgsVectorLayer *layer, QWidget *parent = nullptr );
+    explicit QgsAttributeFormContainerEdit( const QgsAttributesFormTreeData::DnDTreeNodeData &nodeData, QgsVectorLayer *layer, QWidget *parent = nullptr );
+
+    /**
+     * Sets the \a containerName.
+     *
+     * \since QGIS 3.44
+     */
+    void setTitle( const QString &containerName );
```

Comment:
We should avoid referring to title vs name here.

```suggestion
    /**
     * Sets the container \a title.
     *
     * \since QGIS 3.44
     */
    void setTitle( const QString &title );
```